### PR TITLE
Avoid transcoding allocations when serializing doubles

### PR DESCRIPTION
**Pull request by PaulusParssinen**
One of the many low-hanging fruits in SortedSet ops.

### .NET 8

| Method | Runtime  | value                |      Mean | Allocated |
| ------ | -------- | -------------------- | --------: | --------: |
| Before | .NET 8.0 | -1.79(...)E+308 [24] | 185.29 ns |      72 B |
| After  | .NET 8.0 | -1.79(...)E+308 [24] | 181.87 ns |         - |
|        |          |                      |           |           |
| Before | .NET 8.0 | -1                   |  87.59 ns |      32 B |
| After  | .NET 8.0 | -1                   |  75.70 ns |         - |
|        |          |                      |           |           |
| Before | .NET 8.0 | 0                    |  47.84 ns |      24 B |
| After  | .NET 8.0 | 0                    |  42.23 ns |         - |
|        |          |                      |           |           |
| Before | .NET 8.0 | 1                    |  84.80 ns |      24 B |
| After  | .NET 8.0 | 1                    |  74.61 ns |         - |
|        |          |                      |           |           |
| Before | .NET 8.0 | 3.141592653589793    | 148.05 ns |      56 B |
| After  | .NET 8.0 | 3.141592653589793    | 131.54 ns |         - |
|        |          |                      |           |           |
| Before | .NET 8.0 | 1.797(...)E+308 [23] | 182.34 ns |      72 B |
| After  | .NET 8.0 | 1.797(...)E+308 [23] | 179.40 ns |         - |

<details>
<summary>.NET 6</summary>

| Method | Runtime  | value                |      Mean | Allocated |
| ------ | -------- | -------------------- | --------: | --------: |
| Before | .NET 6.0 | -1.79(...)E+308 [24] | 204.37 ns |      72 B |
| After  | .NET 6.0 | -1.79(...)E+308 [24] | 209.13 ns |         - |
|        |          |                      |           |           |
| Before | .NET 6.0 | -1                   | 100.56 ns |      32 B |
| After  | .NET 6.0 | -1                   | 105.70 ns |         - |
|        |          |                      |           |           |
| Before | .NET 6.0 | 0                    |  64.16 ns |      24 B |
| After  | .NET 6.0 | 0                    |  63.50 ns |         - |
|        |          |                      |           |           |
| Before | .NET 6.0 | 1                    |  96.91 ns |      24 B |
| After  | .NET 6.0 | 1                    |  96.01 ns |         - |
|        |          |                      |           |           |
| Before | .NET 6.0 | 3.141592653589793    | 164.66 ns |      56 B |
| After  | .NET 6.0 | 3.141592653589793    | 162.19 ns |         - |
|        |          |                      |           |           |
| Before | .NET 6.0 | 1.797(...)E+308 [23] | 200.08 ns |      72 B |
| After  | .NET 6.0 | 1.797(...)E+308 [23] | 201.84 ns |         - |

</details>

### CODE REVIEW

**Comment by badrishc** [comment]
Great to see the allocation avoidance, seems there might be several other places this can be used.

File: libs/common/RespWriteUtils.cs [hunk 1]
```diff
@@ -435,6 +436,49 @@ public static int GetIntegerAsBulkStringLength(int integer)
...
         }
 
+        /// <summary>
+        /// Try to write a double-precision floating-point <paramref name="value"/> as bulk string.
+        /// </summary>
+        /// <returns><see langword="true"/> if the <paramref name="value"/> could be written to <paramref name="curr"/>; <see langword="false"/> otherwise.</returns>
+        [SkipLocalsInit]
+        public static bool TryWriteDoubleBulkString(double value, ref byte* curr, byte* end)
+        {
+            if (double.IsInfinity(value))
```
Comment (PaulusParssinen) [inline libs/common/RespWriteUtils.cs:446]: We could add basic NaN handling here too if we want to. Nothing exercises these codepaths currently but this logic can later be shared to implement double serialization per RESP3 spec.

Comment (badrishc) [reply libs/common/RespWriteUtils.cs:446]: Yes, that makes sense.

Comment (PaulusParssinen) [reply libs/common/RespWriteUtils.cs]: This subnormal handling is not defined for RESP2 so this is just currently a mix of both RESP2 & RESP3. This is free to change in future, nothing _should_ be getting stored as -Inf, Inf or NaN

**Review by badrishc — APPROVED** [review]

**Pull request merged**